src/main/java/osiris/interpreters/UserInputInterpreter.java

```diff
@@ -0,0 +1,139 @@
+package osiris.interpreters;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import osiris.commands.AddDeadlineTaskCommand;
+import osiris.commands.AddEventTaskCommand;
+import osiris.commands.AddToDoTaskCommand;
+import osiris.commands.Command;
+import osiris.commands.DeleteTaskCommand;
+import osiris.commands.FindTasksCommand;
+import osiris.commands.MarkTaskCompleteCommand;
+import osiris.commands.MarkTaskIncompleteCommand;
+import osiris.commands.NoCommand;
+import osiris.commands.PrintUserTasksCommand;
+import osiris.commands.TerminateChatCommand;
+import osiris.commands.UnsupportedCommand;
+import osiris.exceptions.OsirisException;
+import osiris.formatters.DateTimeFormatters;
+import osiris.validation.InputsValidator;
+
+/**
+ * Interprets user input and returns the appropriate command.
+ */
+public class UserInputInterpreter {
+
+    /** A singleton UserInputInterpreter instance. */
+    private static UserInputInterpreter instance;
+
+    /** Private constructor to initialise an UserInputInterpreter instance. */
+    private UserInputInterpreter() {}
+
+    /**
+     * Returns the singleton instance of the UserInputInterpreter.
+     *
+     * @return The UserInputInterpreter instance.
+     */
+    public static UserInputInterpreter getInstance() {
+        if (instance == null) {
+            instance = new UserInputInterpreter();
+        }
+        return instance;
+    }
+
+    /**
+     * Interprets the user input and returns the corresponding command.
+     *
+     * @param userInput The user input string.
+     * @return The appropriate Command object based on the user input.
+     * @throws OsirisException If there is an error in interpreting data.
+     */
+    public Command interpretUserInput(String userInput) throws OsirisException {
+
+        String[] inputtedWords = userInput.split(" ");
+        String taskName = "";
+        boolean isValid = false;
+
+        switch (inputtedWords[0]) {
+        case TerminateChatCommand.COMMAND:
```

Comment:
I love how the COMMAND field is used to identify specific commands, great work 👍🏼